compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -2125,7 +2132,7 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
             mutableLocalVars += thisSym
 
           val thisLocalIdent = encodeLocalSym(thisSym)
-          val thisLocalType = currentThisType
+          val thisLocalType = currentThisTypeNullable
```

Comment:
Could you drop a comment why the `thisLocalType` needs to be nullable? (I assume it is because of some scalac trees).